src/app/components/students/adding-student-page/adding-student/adding-student.component.ts

```diff
@@ -0,0 +1,218 @@
+import { Component, ViewChild } from '@angular/core';
+import { DataService } from '../../../../common/services/data.service';
+import { Router } from '@angular/router';
+
+import { ModalComponent } from '../../../../shared/components/modal/modal.component';
+
+import { INFO_MESSAGE_FOR_NAME } from '../../../../common/constants/info-message-for-name';
+import { INFO_MESSAGE_FOR_LAST_NAME } from '../../../../common/constants/info-message-for-last-name';
+import { INFO_MESSAGE_FOR_AGE } from '../../../../common/constants/info-message-for-age';
+import { INFO_MESSAGE_FOR_ADDRESS } from '../../../../common/constants/info-message-for-address';
+
+@Component({
+  selector: 'app-adding-student',
+  templateUrl: './adding-student.component.html',
+  styleUrls: ['./adding-student.component.scss']
+})
+export class AddingStudentComponent {
+
+  @ViewChild(ModalComponent, { static: false })
+  private templateModalComponent: ModalComponent;
+
+  name: string = "";
+  lastName: string = "";
+  age: number;
+  address: string = "";
+
+  nameCorrectness: boolean = false;
+  lastNameCorrectness: boolean = false;
+  ageCorrectness: boolean = false;
+  addressCorrectness: boolean = false;
+
+  nameInfo: string = "";
+  lastNameInfo: string = "";
+  ageInfo: string = "";
+  addressInfo: string = "";
+
+  nameRegExp: any = /^[a-zA-Z/\s]+$/;
+  spaceRegExp: any = /\s/;
+  firstLetterUppercaseRegExp: any = /^[A-Z]/;
+  lastNameRegExp: any = /^[a-zA-Z-/\s]+$/;
+  ageRegExp: any = /^[0-9/\s]+$/;
+  addressRegExp: any = /^[a-zA-Z0-9,/\s]+$/;
+
+  messageForName: string[] = INFO_MESSAGE_FOR_NAME;
+  messageForLastName: string[] = INFO_MESSAGE_FOR_LAST_NAME;
+  messageForAge: string[] = INFO_MESSAGE_FOR_AGE;
+  messageForAddress: string[] = INFO_MESSAGE_FOR_ADDRESS;
```

Comment:
using an array, in this case, is bad practise, cause we need to remember position of the messages.